docs/src/locators.md

```diff
@@ -188,104 +186,98 @@ await page.getByRole('checkbox', { checked: true, name: "Check me" }).check();
 ```
 
 ```python async
-await page.locator('data-test-id=directions').click()
+await page.get_by_test_id('directions').click()
 ```
 
 ```python sync
-page.locator('data-test-id=directions').click()
+page.get_by_test_id('directions').click()
 ```
 
 ```csharp
-await page.Locator("data-test-id=directions").ClickAsync();
+await page.GetByTestId("directions").ClickAsync();
 ```
 
-### Locate by label text
+By default, [`method: Page.getByTestId`] will locate elements baed on the `data-test-id` attribute, but you can configure it in your test config or calling [`method: Selectors.setTestIdAttribute`].
+
+### Locate by label text with [`method: Page.getByLabel`]
 
-Most form controls usually have dedicated labels that could be conveniently used to interact with the form. Input actions in Playwright automatically distinguish between labels and controls, so you can just locate the label to perform an action on the associated control.
+Most form controls usually have dedicated labels that could be conveniently used to interact with the form. In this case, you can locate the control by its associated label.
 
 For example, consider the following DOM structure.
 
 ```html
 <label for="password">Password:</label><input type="password">
 ```
 
-You can target the label with something like `text=Password` and perform the following actions on the password input:
-- `click` will click the label and automatically focus the input field;
-- `fill` will fill the input field;
-- `inputValue` will return the value of the input field;
-- `selectText` will select text in the input field;
-- `setInputFiles` will set files for the input field with `type=file`;
-- `selectOption` will select an option from the select box.
-
-For example, to fill the input by targeting the label:
+You can fill the input after locating it by the label text:
 
 ```js
-await page.getByText('Password').fill('secret');
+await page.getByLabel('Password').fill('secret');
 ```
 
 ```java
-page.getByText("Password").fill("secret");
+page.getByLabel("Password").fill("secret");
 ```
 
 ```python async
-await page.get_by_text('Password').fill('secret')
+await page.get_by_label("Password").fill("secret")
 ```
 
 ```python sync
-page.get_by_text('Password').fill('secret')
+page.get_by_label("Password").fill("secret")
 ```
 
 ```csharp
-await page.GetByText("Password").FillAsync("secret");
+await page.GetByLabel("Password").FillAsync("secret");
 ```
 
-However, other methods will target the label itself, for example `textContent` will return the text content of the label, not the input field.
-
 ### Locate in a subtree
 
-You can chain [`method: Page.locator`] and [`method: Locator.locator`] calls to narrow down the search to a particular part of the page.
+You can chain methods that create a locator, like [`method: Page.getByText`] or [`method: Locator.getByRole`], to narrow down the search to a particular part of the page.
 
 For example, consider the following DOM structure:
 
```

Comment:
The example here has data-test-id in the html so if we need to change this to data-testid then keep this in mind.